pkg/odo/genericclioptions/runnable.go

```diff
@@ -77,7 +82,12 @@ func GenericRun(o Runnable, cmd *cobra.Command, args []string) {
 		startTelemetry(cmd, err, startTime)
 	}
 	util.LogErrorAndExit(err, "")
-
+	if cmd.Name() == "push" {
+		// TODO: capture ctrl^z
+		go commonutil.StartSignalWatcher([]os.Signal{syscall.SIGTERM, syscall.SIGQUIT, syscall.SIGINT}, func() {
+			startTelemetry(cmd, errors.Errorf("user quit"), startTime)
```

Comment:
It might be better to actually send the information about signal odo received.

For example adding data like this to telemetry `receivedSignal: SIGTERM`